core/interfaces/i_variable_holder.ts

```diff
@@ -0,0 +1,21 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import type {VariableModel} from '../variable_model.js';
+
+
+/** Interface for a data model that holds a variable model. */
+export interface IVariableHolder {
```

Comment:
I don't love this name. Also, this is only going to used in conjunction with `IParameterModel` right? How about

```
export interface IVariableBackedParameterModel extends IParameterModel {}
```

??

Then in the documentation and/or the plugin we can tell people to implement this one if the parameters will be backed by variables, or the base one if not.